### PR TITLE
[MOL-14960][ES] Add functionality to update search results list title

### PR DESCRIPTION
**Pull request by SimWPEric**
**Changes**

- Add `locationListTitle` prop to support functionality to update search results list title
- Add unit test for `locationListTitle`
- Update location-field stories

**Changelog entry**

-   Ability to change `locationListTitle` from default 'Select location'

**Additional information**

-   You may refer to this [ticket](https://jira.ship.gov.sg/browse/MOL-14960)

### CODE REVIEW

**Review by ryhuz — DISMISSED** [review]

File: src/components/fields/location-field/location-modal/types.ts [hunk 1]
```diff
@@ -19,4 +19,5 @@ export interface ILocationModalProps
 	onClose: () => void;
 	onConfirm: (values: ILocationFieldValues) => void;
 	updateFormValues: (values: ILocationFieldValues, shouldDirty?: boolean) => void;
+	locationListTitle?: string;
```
Comment (weili-govtech) [inline src/components/fields/location-field/location-modal/types.ts:22]: - optional keys should be a union with undefined to support projects using exactOptionalPropertyTypes in tsconfig

```suggestion
	locationListTitle?: string | undefined;
```

File: src/components/fields/location-field/types.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export interface ILocationFieldSchema<V = undefined>
 		Pick<IStaticMapProps, "staticMapPinColor"> {
 	className?: string;
 	locationModalStyles?: string | undefined;
+	locationListTitle?: string;
```
Comment (weili-govtech) [inline src/components/fields/location-field/types.ts:21]:
```suggestion
	locationListTitle?: string | undefined;
```

File: src/components/fields/location-field/location-field.tsx [hunk 3]
```diff
@@ -35,6 +35,7 @@ export const LocationField = (props: IGenericFieldProps<ILocationFieldSchema>) =
 			reverseGeoCodeEndpoint,
 			staticMapPinColor,
 			validation,
+			locationListTitle,
```
Comment (weili-govtech) [inline src/components/fields/location-field/location-field.tsx:38]: sort destructured props by alphabetical order due to the sheer amount of props destructured

File: src/__tests__/components/fields/location-field/location-field.spec.tsx [hunk 4]
```diff
@@ -1415,4 +1415,56 @@ describe("location-input-group", () => {
...
+				onSuccess(mock1PageFetchAddressResponse);
+			});
+			fetchSingleLocationByLatLngSpy.mockImplementation(
+				(_reverseGeoCodeEndpoint, _convertLatLngToXYEndpoint, _lat, _lng, handleResult) => {
+					handleResult(fetchSingleLocationByLatLngSingleReponse);
+				}
+			);
+		});
+
+		it("show default location list title", () => {
```
Comment (weili-govtech) [inline src/__tests__/components/fields/location-field/location-field.spec.tsx:1431]:
```suggestion
		it("should show default location list title if locationListTitle is not provided", () => {
```

File: src/__tests__/components/fields/location-field/location-field.spec.tsx [hunk 5]
```diff
@@ -1415,4 +1415,56 @@ describe("location-input-group", () => {
...
+				},
+				overrideField: {
+					reverseGeoCodeEndpoint: "https://www.mock.com/reverse-geo-code",
+				},
+			});
+			const locationListTitle = screen.getByText("Select location");
+			expect(locationListTitle).toBeInTheDocument();
+		});
+
+		it("show updated location list title", () => {
```
Comment (weili-govtech) [inline src/__tests__/components/fields/location-field/location-field.spec.tsx:1450]:
```suggestion
		it("should show location list title according to locationListTitle", () => {
```

File: src/__tests__/components/fields/location-field/location-field.spec.tsx [hunk 6]
```diff
@@ -1415,4 +1415,56 @@ describe("location-input-group", () => {
 			expect(formIsDirty).toBe(false);
 		});
 	});
+
+	describe("search results list title", () => {
```
Comment (weili-govtech) [inline src/__tests__/components/fields/location-field/location-field.spec.tsx:1419]: both tests are failing if they are the only one running through `it.only`

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/__tests__/components/fields/location-field/location-field.spec.tsx (hunk 6 above)
Comment (SimWPEric) [reply src/__tests__/components/fields/location-field/location-field.spec.tsx:1419]: Added `async await` for both test cases, should be passing when running through `it.only`

File: src/components/fields/location-field/types.ts (hunk 2 above)
Comment (SimWPEric) [reply src/components/fields/location-field/types.ts]: Resolved

**Review by weili-govtech — DISMISSED** [review]

**Comment by weili-govtech** [comment]
need to resolve merge conflicts

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**